connections/sim/hw/hw_sim.py

```diff
@@ -35,6 +36,10 @@ def __init__(
 
         self._sensors = {s.get_type(): s for s in sensors}
 
+        self._voltage_sensor = None
+        if SensorType.VOLTAGE in self._sensors:
+            self._voltage_sensor = self._sensors[SensorType.VOLTAGE]
```

Comment:
If you don't use `self._voltage_sensor` and just call `self.sensor_read` when handling the read, then you don't need any special replace sensor logic either.